liquidctl/driver/nzxt_smart_device.py

```diff
@@ -182,66 +272,188 @@ def get_status(self, **kwargs):
+        'backwards-moving-alternating':  (0x05, 0x18, 0x00, 2, 2),
+        'pulse':                         (0x06, 0x00, 0x00, 1, 8),
+        'breathing':                     (0x07, 0x00, 0x00, 1, 8),   # colors for each step
+        'super-breathing':               (0x03, 0x19, 0x00, 1, 40),  # independent leds
+        'starry-night':                  (0x09, 0x00, 0x00, 1, 1),
+        'rainbow-flow':                  (0x0b, 0x00, 0x00, 0, 0),
+        'super-rainbow':                 (0x0c, 0x00, 0x00, 0, 0),   
+        'rainbow-pulse':                 (0x0d, 0x00, 0x00, 0, 0),
+        'backwards-rainbow-flow':        (0x0b, 0x01, 0x00, 0, 0),
+        'backwards-super-rainbow':       (0x0c, 0x01, 0x00, 0, 0),   
+        'backwards-rainbow-pulse':       (0x0d, 0x01, 0x00, 0, 0),
+    }
+
+    def __init__(self, device, description, speed_channel_count, color_channel_count, **kwargs):
+        """Instantiate a driver with a device handle."""
+        speed_channels = {'fan{}'.format(i + 1): (i, _MIN_DUTY, _MAX_DUTY)
+                          for i in range(speed_channel_count)}
+        color_channels = {'led{}'.format(i + 1): (i)
+                          for i in range(color_channel_count)}
+        super().__init__(device, description, speed_channels, color_channels, **kwargs)
+
+    def initialize(self, **kwargs):
+        """Initialize the device.
+
+        Detects all connected fans and LED accessories, and allows subsequent
+        calls to get_status.
+        """
+        self._write([0x60, 0x02, 0x01, 0xE8, 0x03, 0x01, 0xE8, 0x03])
         self.device.release()
 
-    def set_fixed_speed(self, channel, duty, **kwargs):
-        """Set channel to a fixed speed."""
-        if channel == 'sync':
-            selected_channels = self._speed_channels
-        else:
-            selected_channels = { channel: self._speed_channels[channel] }
-        for cname, (cid, smin, smax) in selected_channels.items():
-            if duty < smin:
-                duty = smin
-            elif duty > smax:
-                duty = smax
-            LOGGER.info('setting %s duty to %i%%', cname, duty)
-            self._write([0x2, 0x4d, cid, 0, duty])
+    def get_status(self, **kwargs):
+        """Get a status report.
+
+        Returns a list of (key, value, unit) tuples.
+        """
+        status = []
+        num_valid_replies_recvd = 0
+        msg_1101_reply = False
+        msg_2103_reply = False
+        msg_6702_reply = False
+        msg_6704_reply = False
+        accessory_name_dict = {
+            0x04: "HUE 2 LED Strip",
+            0x08: "HUE 2 Cable Comb", 
+            0x0a: "HUE 2 Underglow 200mm",
+            0x0b: "AER RGB 2 120mm", 
+            0x0c: "AER RGB 2 140mm"
+        }
+        # Get configuration information from device
+        LOGGER.debug('Issuing command 0x10 0x01 to get firmware info')
+        self._write([0x10, 0x01, 0x00, 0x00, 0x00, 0x00])
```

Comment:
The trailing `0x00` bytes are not necessary, as `_write` will pad the message to `_WRITE_LENGTH` bytes with zeros.  But you can keep them if you think that they improve legibility.

Reply:
Done